sys/kern/vmem.c

```diff
@@ -29,45 +32,111 @@ typedef struct vmem {
 
 typedef enum { BT_TYPE_FREE, BT_TYPE_BUSY, BT_TYPE_SPAN } bt_type_t;
 
-typedef struct vmem_bt {
-  TAILQ_ENTRY(vmem_bt) bt_seglist;
+typedef struct bt {
+  TAILQ_ENTRY(bt) bt_seglist;
   union {
-    LIST_ENTRY(vmem_bt) u_freelist;
-    LIST_ENTRY(vmem_bt) u_hashlist;
+    LIST_ENTRY(bt) u_freelist;
+    LIST_ENTRY(bt) u_hashlist;
   } bt_u;
```

Comment:
We can use anonymous unions in C11. You don't need macro definitions below.

Reply:
Oh, I see, and it would simplify the code. Thanks!